src/field_map_change.c

```diff
@@ -180,57 +180,57 @@ static const UnkStruct_ov61_0222C884 Unk_020EC3A0 = {
 
 static void sub_020530C8 (FieldSystem * fieldSystem)
 {
-    BOOL v0;
+    BOOL inBattleTower;
 
-    switch (fieldSystem->unk_1C->unk_00) {
+    switch (fieldSystem->location->mapId) {
     case 326:
     case 327:
     case 328:
     case 329:
     case 330:
     case 331:
     case 493:
```

Comment:
Don't we have header IDs for these? Or are these specifically maps and not headers?

Reply:
These are specifically map IDs, couldn't find where they were if we do have consts for them

Reply:
https://github.com/pret/pokeplatinum/blob/main/include/data/map_headers.h

I checked some of these in the binary tool, these are in fact just header IDs. e.g. `326` maps to `MAP_HEADER_BATTLE_TOWER`, 331 to `MAP_HEADER_BATTLE_TOWER_MULTI_BATTLE_ROOM`, etc.